### PR TITLE
feat: add a wallet sign command

### PR DESCRIPTION
**Pull request by edpoptart**
Executes the ``sign`` command to sign a message with the provided wallet or wallet hotkey.

    This command signs a message using the provided wallet.

    Usage:
        The command generates a signature for a given message using the provided wallet

    Optional arguments:
        --wallet.hotkey: A hotkey to use to sign the message instead of the wallet coldkey

    Example usage::

        btcli wallet sign --wallet.name default --message '{"something": "here", "timestamp": 1719908486}'
        btcli wallet sign --wallet.name default --wallet.name hotkey --message '{"something": "here", "timestamp": 1719908486}'

    Note:
        When using `btcli`, `w` is used interchangeably with `wallet`. You may use either based on your preference for brevity or clarity.
        This command is essential for users to easily prove their ownership over a coldkey or a hotkey.

### CODE REVIEW

**Review by Rapiiidooo — APPROVED** [review]

**Review by alex-drocks — APPROVED** [review]

**Review by alex-drocks — APPROVED** [review]

File: bittensor/commands/wallets.py [hunk 1]
```diff
@@ -1099,3 +1099,81 @@ def create_transfer_history_table(transfers):
...
+    Note:
+        When using `btcli`, `w` is used interchangeably with `wallet`. You may use either based on your preference for brevity or clarity.
+        This command is essential for users to easily prove their ownership over a coldkey or a hotkey.
+    """
+
+    @staticmethod
+    def run(cli):
+        r"""Sign a message using the provided wallet or hotkey."""
+        wallet = bittensor.wallet(config=cli.config)
+        keypair = wallet.coldkey
```
Comment (thealligatorking) [inline bittensor/commands/wallets.py:1138]: for unlocking the coldkey, could you wrap 

```
        try:
            wallet.coldkey
        except bittensor.KeyFileError:
            bittensor.__console__.print(
                ":cross_mark: [red]Keyfile is corrupt, non-writable, non-readable or the password used to decrypt is invalid[/red]:[bold white]\n  [/bold white]"
            )
```

**Comment by thealligatorking** [comment]
This looks great! One comment.

File: bittensor/commands/wallets.py (hunk 1 above)
Comment (edpoptart) [reply bittensor/commands/wallets.py:1138]: Sure I'll wrap it.

**Comment by edpoptart** [comment]
Should I add a check to make sure the user isn't trying to sign an extrinsic? In case a malicious actor tells a user to sign something --which would turn out to be a transfer extrinsic -- to prove that they own the wallet.

Edit: Since the message passed to the command is a string, a user would not be able to pass an extrinsic as a message. I have been unable to decode an extrinsic to string directly, due to the way its data is wrapped

File: bittensor/commands/wallets.py [hunk 2]
```diff
@@ -1099,3 +1099,81 @@ def create_transfer_history_table(transfers):
...
+    Usage:
+        The command generates a signature for a given message using the provided wallet
+
+    Optional arguments:
+        --wallet.hotkey: A hotkey to use to sign the message instead of the wallet coldkey
+
+    Example usage::
+
+        btcli wallet sign --wallet.name default --message '{"something": "here", "timestamp": 1719908486}'
+        btcli wallet sign --wallet.name default --wallet.name hotkey --message '{"something": "here", "timestamp": 1719908486}'
```
Comment (camfairchild) [inline bittensor/commands/wallets.py]:
```suggestion
        btcli wallet sign --wallet.name default --wallet.hotkey hotkey --message '{"something": "here", "timestamp": 1719908486}'
```

**Comment by alex-drocks** [comment]
Lgtm

**Review by IvanAnishchuk — APPROVED** [review]
LGTM!

I think this feature is important and standard for most Crypto ecosystems. There's interest from several subnet operators and validators. Can we please get this rolling?

Is there anything anybody can do to help merge this quickly?

**Comment by thewhaleking** [comment]
This was rolled into the new [btcli](https://github.com/opentensor/btcli)

**Pull request closed without merging**